PACKAGE = "color_detector_params"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("only_publish_mask", bool_t, 0, "do not calc target clustering", False)
gen.add("publish_target", bool_t, 0, "do not calc target clustering", False)
gen.add("blue_enable", bool_t, 0, "do not calc target clustering", False)
gen.add("red_enable", bool_t, 0, "do not calc target clustering", False)
gen.add("green_enable", bool_t, 0, "do not calc target clustering", False)
gen.add("yellow_enable", bool_t, 0, "do not calc target clustering", False)
gen.add("orange_enable", bool_t, 0, "do not calc target clustering", False)
gen.add("purple_enable", bool_t, 0, "do not calc target clustering", False)

gen.add("HIGHEST_TARGET_Y", double_t, 0, "discription", 0.0, -2.0, 2.0)
gen.add("LOWEREST_TARGET_Y", double_t, 0, "discription", 0.0, -2.0, 2.0)

gen.add("LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA1_LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA1_LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA1_LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA1_LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA1_LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA1_LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA1_UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA1_UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA2_LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA2_LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA2_LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA2_LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA2_LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA2_LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA2_UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA2_UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA3_LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA3_LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA3_LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA3_LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA3_LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA3_LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA3_UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA3_UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA4_LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA4_LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA4_LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA4_LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA4_LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA4_LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA4_UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA4_UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA5_LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA5_LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA5_LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA5_LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA5_LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA5_LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA5_UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA5_UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA6_LOWER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_LOWER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_LOWER_RED_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_RED_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_UPPER_RED_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_RED_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA6_LOWER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_LOWER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_LOWER_GREEN_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_GREEN_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_UPPER_GREEN_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_GREEN_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA6_LOWER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_LOWER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_LOWER_BLUE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_BLUE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_UPPER_BLUE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_BLUE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA6_LOWER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_LOWER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_LOWER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_YELLOW_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_UPPER_YELLOW_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_YELLOW_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA6_LOWER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_LOWER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_LOWER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_ORANGE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_UPPER_ORANGE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_ORANGE_V", int_t, 0, "discription", 0, 0, 255)

gen.add("ROOMBA6_LOWER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_LOWER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_LOWER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_PURPLE_H", int_t, 0, "discription", 0, 0, 180)
gen.add("ROOMBA6_UPPER_PURPLE_S", int_t, 0, "discription", 0, 0, 255)
gen.add("ROOMBA6_UPPER_PURPLE_V", int_t, 0, "discription", 0, 0, 255)

exit(gen.generate(PACKAGE, "color_detector_params", "Hsv"))
